hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestPartitionStatsIndex.scala

```diff
@@ -224,6 +226,126 @@ class TestPartitionStatsIndex extends PartitionStatsIndexTestBase {
     assertEquals(67, snapshot0.count())
   }
 
+  /**
+   * Test case to do updates and then validate partition stats with cleaning.
+   */
+  @ParameterizedTest
+  @EnumSource(classOf[HoodieTableType])
+  def testPartitionStatsWithCompactionAndCleaning(tableType: HoodieTableType): Unit = {
+    var hudiOpts = commonOpts ++ Map(
+      DataSourceWriteOptions.TABLE_TYPE.key -> tableType.name(),
+      HoodieCleanConfig.CLEANER_COMMITS_RETAINED.key() -> "1")
+    if (tableType == HoodieTableType.MERGE_ON_READ) {
+      hudiOpts = hudiOpts ++ Map(
+        HoodieCompactionConfig.INLINE_COMPACT.key() -> "true",
+        HoodieCompactionConfig.INLINE_COMPACT_NUM_DELTA_COMMITS.key() -> "2",
+        HoodieCompactionConfig.PARQUET_SMALL_FILE_LIMIT.key() -> "0",
+        HoodieMetadataConfig.COMPACT_NUM_DELTA_COMMITS.key() -> "15"
+      )
+    }
+    // insert followed by two upserts (trigger a compaction so that prev version can be cleaned)
+    doWriteAndValidateDataAndPartitionStats(
+      hudiOpts,
+      operation = DataSourceWriteOptions.INSERT_OPERATION_OPT_VAL,
+      saveMode = SaveMode.Overwrite)
+    doWriteAndValidateDataAndPartitionStats(
+      hudiOpts,
+      operation = DataSourceWriteOptions.UPSERT_OPERATION_OPT_VAL,
+      saveMode = SaveMode.Append)
+    doWriteAndValidateDataAndPartitionStats(
+      hudiOpts,
+      operation = DataSourceWriteOptions.UPSERT_OPERATION_OPT_VAL,
+      saveMode = SaveMode.Append)
+    // Clean Operation
+    val lastCleanInstant = getLatestMetaClient(false).getActiveTimeline.getCleanerTimeline.lastInstant()
```

Comment:
Add a validation that the compaction commit is present?

Reply:
done